### ISSUE TITLE
make: *** [mnist8m.model] Error 139

### ISSUE THREAD

**Issue by mrgloom** [JohnLangford/vowpal_wabbit#744]
Seems mnist8m dataset is downloadable now.
https://github.com/JohnLangford/vowpal_wabbit/issues/541

But I get another error `make: *** [mnist8m.model] Error 139`

`gcc --version`
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3

make mnist8m.test.confusion
./do-mnist8m-train
final_regressor = mnist8m.model
Num weight bits = 24
learning rate = 0.1
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = 
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0        6        1      167
1.000000 1.000000            2            2.0        1        6      177
1.000000 1.000000            4            4.0        2        6       97
1.000000 1.000000            8            8.0        4        2      201
0.812500 0.625000           16           16.0        8        7      145
0.843750 0.875000           32           32.0        9        2      176
0.671875 0.500000           64           64.0        1        1      241
0.460938 0.250000          128          128.0        5        5      151
0.390625 0.320312          256          256.0        4        4      189
0.306641 0.222656          512          512.0        8        8      135
0.240234 0.173828         1024         1024.0        8        8       97
0.196777 0.153320         2048         2048.0        2        2       93
0.151123 0.105469         4096         4096.0       10       10      145
0.130615 0.110107         8192         8192.0        7        7      135
0.118103 0.105591        16384        16384.0        8        8      151
0.098785 0.079468        32768        32768.0       10       10      106
0.083435 0.068085        65536        65536.0        8        8      180
../../vowpalwabbit/vw --oaa 10 -f mnist8m.model -b 24 --adaptive --invariant   19.93s user 6.96s system 110% cpu 24.254 total
make: *** [mnist8m.model] Error 139

**Comment by arielf** [comment]
Is the output above missing parts?

`vw` output seems to be cut in the middle of training.

Error 139 from make usually indicates a SEGV while compiling with gcc [http://www.tldp.org/FAQ/Linux-FAQ/error-messages.html#make-says] - but the output above doesn't show any compile line.

And I can't reproduce this on Ubuntu 14.04 (works just fine for me), which usually indicates a problem elsewhere in your env.  Please make sure to clean and start with no old binaries, left-over models from older version runs, etc.  If you still get error 139 in make, your compiler may be a buggy version.  Try to run the last `gcc` command that fails in isolation (not via make) and see if it causes a SEGV.

For reference, here's my full output with latest:

```
demo/mnist> make mnist8m.test.confusion
wget http://leon.bottou.org/_media/projects/infimnist.tar.gz
--2015-08-11 22:13:25--  http://leon.bottou.org/_media/projects/infimnist.tar.gz
Resolving leon.bottou.org (leon.bottou.org)... 128.122.47.89
Connecting to leon.bottou.org (leon.bottou.org)|128.122.47.89|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 365872710 (349M) [application/octet-stream]
Saving to: 'infimnist.tar.gz'

100%[======================================>] 365,872,710 6.59MB/s   in 52s    

2015-08-11 22:14:18 (6.66 MB/s) - 'infimnist.tar.gz' saved [365872710/365872710]

tar xvfz infimnist.tar.gz
infimnist/
infimnist/.gitattributes
infimnist/.gitignore
infimnist/COPYING
infimnist/Makefile
infimnist/NMakefile
infimnist/README
infimnist/data/
infimnist/data/fields_float_1522x28x28.bin
infimnist/data/t10k-images-idx3-ubyte
infimnist/data/t10k-labels-idx1-ubyte
infimnist/data/tangVec_float_60000x28x28.bin
infimnist/data/train-images-idx3-ubyte
infimnist/data/train-labels-idx1-ubyte
infimnist/infimnist.c
infimnist/infimnist.h
infimnist/main.c
make -C infimnist
make[1]: Entering directory `.../src/vowpal_wabbit/arielf/demo/mnist/infimnist'
cc -O3 -Wall -c main.c
cc -O3 -Wall -c infimnist.c
cc -O3 -Wall -o infimnist main.o infimnist.o 
make[1]: Leaving directory `.../src/vowpal_wabbit/arielf/demo/mnist/infimnist'
(cd infimnist && ./infimnist lab 10000 8109999) | gzip > train8m-labels-idx1-ubyte.gz
(cd infimnist && ./infimnist pat 10000 8109999) | gzip > train8m-images-idx3-ubyte.gz
./do-mnist8m-train
final_regressor = mnist8m.model
Num weight bits = 24
learning rate = 0.1
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = 
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0        6        1      167
1.000000 1.000000            2            2.0        1        6      177
1.000000 1.000000            4            4.0        2        6       97
1.000000 1.000000            8            8.0        4        2      201
0.812500 0.625000           16           16.0        8        7      145
0.843750 0.875000           32           32.0        9        2      176
0.671875 0.500000           64           64.0        1        1      241
0.460938 0.250000          128          128.0        5        5      151
0.390625 0.320312          256          256.0        4        4      189
0.306641 0.222656          512          512.0        8        8      135
0.240234 0.173828         1024         1024.0        8        8       97
0.196777 0.153320         2048         2048.0        2        2       93
0.151123 0.105469         4096         4096.0       10       10      145
0.130615 0.110107         8192         8192.0        7        7      135
0.118103 0.105591        16384        16384.0        8        8      151
0.098785 0.079468        32768        32768.0       10       10      106
0.083435 0.068085        65536        65536.0        8        8      180
0.077347 0.071259       131072       131072.0        1        1      300
0.062122 0.046898       262144       262144.0        1        1      253
0.046875 0.031628       524288       524288.0        7        7      184
0.033880 0.020885      1048576      1048576.0        8        8      146
0.023246 0.012611      2097152      2097152.0        3        3      243
0.014761 0.006277      4194304      4194304.0        1        1      224

finished run
number of examples per pass = 8100000
passes used = 1
weighted example sum = 8100000.000000
weighted label sum = 0.000000
average loss = 0.008744
total feature number = 1620363671
../../vowpalwabbit/vw --oaa 10 -f mnist8m.model -b 24 --adaptive --invariant   2148.05s user 16.65s system 113% cpu 31:55.21 total
./do-test mnist8m
only testing
predictions = mnist8m.test.predictions
Num weight bits = 24
learning rate = 10
initial_t = 1
power_t = 0.5
using no cache
Reading datafile =
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0        8        8      117
0.000000 0.000000            2            2.0        3        3      166
0.000000 0.000000            4            4.0        1        1      194
0.000000 0.000000            8            8.0       10       10      130
0.000000 0.000000           16           16.0        6        6      138
0.031250 0.062500           32           32.0        2        2       62
0.015625 0.000000           64           64.0        4        4      138
0.007812 0.000000          128          128.0        6        6      160
0.015625 0.023438          256          256.0        8        8      104
0.013672 0.011719          512          512.0        5        5      100
0.017578 0.021484         1024         1024.0        5        5      156
0.022949 0.028320         2048         2048.0        7        7      193
0.023682 0.024414         4096         4096.0       10       10      157
0.016724 0.009766         8192         8192.0        7        7      172

finished run
number of examples per pass = 10000
passes used = 1
weighted example sum = 10000.000000
weighted label sum = 0.000000
average loss = 0.015800
total feature number = 1521219
mnist8m test errors: 158 out of 10000
confusion matrix (rows = truth, columns = prediction):
975     2       4       5       1       6       4       3       7       5
0       1128    1       0       0       0       1       1       0       2
1       4       1015    3       1       0       0       10      2       0
0       1       0       994     0       5       1       2       0       1
1       0       0       0       972     1       1       1       5       10
1       0       0       5       0       874     3       0       3       3
0       0       0       0       5       2       947     0       1       0
1       0       9       0       1       1       0       1006    4       4
0       0       1       1       0       2       1       3       949     2
1       0       2       2       2       1       0       2       3       982
```

**Comment by mrgloom** [comment]
Seems it was second run, after dataset was downloaded.

Yes, I google for `Error 139` one of the possible reasons is `old version of GCC` so I posteed  output of  `gcc --version`. Also I use `Ubuntu 12.04.5 LTS`.

I get the latest version from github via `git clone` and seems compilation via `make` runs smoothly.
Only one warning:

cb_adf.cc: In function ‘void CB_ADF::get_observed_cost(cb_adf&, v_array<example*>&)’:
cb_adf.cc:181:5: warning: ‘cl’ may be used uninitialized in this function [-Wuninitialized]

Also I run `make test` and all test passed.


Here is full log:

```
make mnist8m.test.confusion
wget http://leon.bottou.org/_media/projects/infimnist.tar.gz
--2015-08-12 11:43:00--  http://leon.bottou.org/_media/projects/infimnist.tar.gz
Resolving leon.bottou.org (leon.bottou.org)... 128.122.47.89
Connecting to leon.bottou.org (leon.bottou.org)|128.122.47.89|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 365872710 (349M) [application/octet-stream]
Saving to: `infimnist.tar.gz'

100%[======================================>] 365,872,710  582K/s   in 10m 38s 

2015-08-12 11:53:39 (560 KB/s) - `infimnist.tar.gz' saved [365872710/365872710]

tar xvfz infimnist.tar.gz
infimnist/
infimnist/.gitattributes
infimnist/.gitignore
infimnist/COPYING
infimnist/Makefile
infimnist/NMakefile
infimnist/README
infimnist/data/
infimnist/data/fields_float_1522x28x28.bin
infimnist/data/t10k-images-idx3-ubyte
infimnist/data/t10k-labels-idx1-ubyte
infimnist/data/tangVec_float_60000x28x28.bin
infimnist/data/train-images-idx3-ubyte
infimnist/data/train-labels-idx1-ubyte
infimnist/infimnist.c
infimnist/infimnist.h
infimnist/main.c
make -C infimnist
make[1]: Entering directory `/some_path/vowpal_wabbit/demo/mnist/infimnist'
cc -O3 -Wall -c main.c
cc -O3 -Wall -c infimnist.c
cc -O3 -Wall -o infimnist main.o infimnist.o 
make[1]: Leaving directory `/media/sf_shared_with_VM/yandex_server/vowpal_wabbit/demo/mnist/infimnist'
(cd infimnist && ./infimnist lab 10000 8109999) | gzip > train8m-labels-idx1-ubyte.gz
(cd infimnist && ./infimnist pat 10000 8109999) | gzip > train8m-images-idx3-ubyte.gz
g++ -O3 -Wall    extractfeatures.cpp   -o extractfeatures
./do-mnist8m-train
final_regressor = mnist8m.model
Num weight bits = 24
learning rate = 0.1
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = 
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0        6        1      167
1.000000 1.000000            2            2.0        1        6      177
1.000000 1.000000            4            4.0        2        6       97
1.000000 1.000000            8            8.0        4        2      201
0.812500 0.625000           16           16.0        8        7      145
0.843750 0.875000           32           32.0        9        2      176
0.671875 0.500000           64           64.0        1        1      241
0.460938 0.250000          128          128.0        5        5      151
0.390625 0.320312          256          256.0        4        4      189
0.306641 0.222656          512          512.0        8        8      135
0.240234 0.173828         1024         1024.0        8        8       97
0.196777 0.153320         2048         2048.0        2        2       93
0.151123 0.105469         4096         4096.0       10       10      145
0.130615 0.110107         8192         8192.0        7        7      135
0.118103 0.105591        16384        16384.0        8        8      151
0.098785 0.079468        32768        32768.0       10       10      106
../../vowpalwabbit/vw --oaa 10 -f mnist8m.model -b 24 --adaptive --invariant   8.86s user 3.49s system 109% cpu 11.298 total
make: *** [mnist8m.model] Error 139

```

**Comment by JohnLangford** [comment]
Unfortunately, I cannot reproduce on my local machine (ubuntu 14.04).

-John

**Comment by mrgloom** [comment]
Finally I test it on fresh Ubuntu 14.04 on VirtualBox and it works, and then on fresh Ubuntu 12.04 on VirtualBox and it's also works, for some reason it still don't work on my old ubuntu 12.04(also on VirtualBox) even after I have updated compiler.

So, issue can be closed, something broken on my side.
